kratos/spatial_containers/specialized_spatial_search.h

```diff
@@ -0,0 +1,807 @@
+        const RadiusArrayType& rRadius,
+        VectorResultConditionsContainerType& rResults,
+        VectorDistanceType& rResultsDistance
+        ) override;
+
+    /**
+    * @brief Search neighbours for every Condition in "rInputConditions" excluding itself
+    * @param rStructureConditions   List of Conditions against which the neighbours are searched
+    * @param rInputConditions       List of Conditions to be searched
+    * @param rRadius                List of search radius for every Condition
+    * @param rResults               Array of results for each Condition
+    */
+    void SearchConditionsInRadiusExclusive(
+        const ConditionsContainerType& rStructureConditions,
+        const ConditionsContainerType& rInputConditions,
+        const RadiusArrayType& rRadius,
+        VectorResultConditionsContainerType& rResults
+        ) override;
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        std::stringstream buffer;
+        buffer << "SpecializedSpatialSearch" ;
+
+        return buffer.str();
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << "SpecializedSpatialSearch";
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+
+    }
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    Parameters mParameters; /// The configuration parameters
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    /**
+     * @brief This method returns the default parameters
+     * @return The default parameters
+     */
+    Parameters GetDefaultParameters() const;
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+private:
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+    /**
+     * @brief This method prepares the search
+     * @param rStructure The structure to be searched
+     * @param rInput The input to be searched
+     * @param rResults The results
+     * @param rResultsDistance The results distance
+     * @tparam TContainer The container type
+     * @tparam TResultType The result type
+     */
+    template<class TContainer, class TResultType>
+    std::vector<typename PointObject<typename TContainer::value_type>::Pointer> PrepareSearch(
```

Comment:
why not just collect the points themselves in the vector? Storing the pointers basically reduces the efficiency to that of a linked list.

Reply:
(therefore not my choise)